script.transmission/resources/lib/transmissionrpc/constants.py

```diff
@@ -1,295 +1,572 @@
-        "speed-limit-down-enabled":     ('boolean', 1, None, None, None, 'Enables download speed limiting.'),
-        "speed-limit-up":               ('number', 1, None, None, None, 'Upload speed limit (in Kib/s).'),
-        "speed-limit-up-enabled":       ('boolean', 1, None, None, None, 'Enables upload speed limiting.'),
-        "start-added-torrents":         ('boolean', 9, None, None, None, 'Added torrents will be started right away.'),
-        "trash-original-torrent-files": ('boolean', 9, None, None, None, 'The .torrent file of added torrents will be deleted.'),
-        'utp-enabled':                  ('boolean', 13, None, None, None, 'Enables Micro Transport Protocol (UTP).'),
-    },
-}
+# -*- coding: utf-8 -*-
+# Copyright (c) 2008-2013 Erik Svensson <erik.public@gmail.com>
+# Licensed under the MIT license.
+
+import logging
+from six import iteritems
+
+LOGGER = logging.getLogger('transmissionrpc')
+LOGGER.setLevel(logging.ERROR)
+
+
+def mirror_dict(source):
+    """
+    Creates a dictionary with all values as keys and all keys as values.
+    """
+    source.update(dict((value, key) for key, value in iteritems(source)))
+    return source
+
+
+DEFAULT_PORT = 9091
+
+DEFAULT_TIMEOUT = 30.0
+
+TR_PRI_LOW = -1
+TR_PRI_NORMAL = 0
+TR_PRI_HIGH = 1
+
+PRIORITY = mirror_dict({
+    'low': TR_PRI_LOW,
+    'normal': TR_PRI_NORMAL,
+    'high': TR_PRI_HIGH
+})
+
+TR_RATIOLIMIT_GLOBAL = 0  # follow the global settings
+TR_RATIOLIMIT_SINGLE = 1  # override the global settings, seeding until a certain ratio
+TR_RATIOLIMIT_UNLIMITED = 2  # override the global settings, seeding regardless of ratio
+
+RATIO_LIMIT = mirror_dict({
+    'global': TR_RATIOLIMIT_GLOBAL,
+    'single': TR_RATIOLIMIT_SINGLE,
+    'unlimited': TR_RATIOLIMIT_UNLIMITED
+})
+
+TR_IDLELIMIT_GLOBAL = 0  # follow the global settings
+TR_IDLELIMIT_SINGLE = 1  # override the global settings, seeding until a certain idle time
+TR_IDLELIMIT_UNLIMITED = 2  # override the global settings, seeding regardless of activity
+
+IDLE_LIMIT = mirror_dict({
+    'global': TR_RATIOLIMIT_GLOBAL,
+    'single': TR_RATIOLIMIT_SINGLE,
+    'unlimited': TR_RATIOLIMIT_UNLIMITED
+})
+
+# A note on argument maps
+# These maps are used to verify *-set methods. The information is structured in
+# a tree.
+# set +- <argument1> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#  |  +- <argument2> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#  |
+# get +- <argument1> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#     +- <argument2> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+
+# Arguments for torrent methods
+TORRENT_ARGS = {
+    'get': {
+        'activityDate': ('number', 1, None, None, None,
```

Comment:
I'm assuming the strings here are only used internally, but if they are displayed to the end user via the GUI, they need to be localized.

Reply:
These are JSON-RPC method arguments :)